File: src/java_exericse/Exercise274.java
package java_exericse;


/**
 * 274. H 指数
 */

import java.util.Arrays;
public class Exercise274 {
    public int hIndex(int[] citations) {
        Arrays.sort(citations);
        int ans = 0;
        for(int i = citations.length-1; i>=0 ;i--){
            if(citations[i]>= citations.length-i){
                ans = citations.length - i;
            }
        }
        return ans;
    }
}

File: src/kotlin_exercise/Exercise274.kt
package kotlin_exercise


/**
 *274. H 指数
 */

class Exercise274{

    fun hIndex(citations: IntArray): Int {
        citations.sort()
        var ans = 0
        for (i in citations.size-1 downTo 0){
            if (citations[i] >= citations.size - i){
                ans = citations.size - i;
            }
        }
        return ans
    }
}